### PR TITLE
Change: Switch inform_mode reports to DEBUG|DEBUG_bundlename

### PR DESCRIPTION
**Pull request by nickanderson**
Inform mode is intended to show changes to the system. The current use of
inform_mode throughout the framework can lead to much more information than
needed.

Using a DEBUG|DEBUG_bundlename class guard allows this policy level debug
information to be reported for the entire policy run or for individual bundles
in a consistent way.

Ref: https://dev.cfengine.com/issues/7197

Superceededs https://github.com/cfengine/masterfiles/pull/464

### CODE REVIEW

**Comment by tzz** [comment]
YES

**Comment by nickanderson** [comment]
trigger build

**Comment by kacfengine** [comment]
Needs rebasing.

**Comment by nickanderson** [comment]
trigger build

**Comment by nickanderson** [comment]
@kacfengine rebased

**Comment by kacfengine** [comment]
Well, that's a juicy one, the policy crashes cf-agent all over the place! For example: http://jenkins.usdc.cfengine.com/view/Pull%20requests/job/testing-enterprise-pr_masterfiles/36/label=PACKAGES_ppc64_aix_5_3/testReport/junit/._17_packages_11_old_unsafe_named_pkg_present/cf/named_pkg_present_cf/

**Comment by nickanderson** [comment]
Embarrassing! thats what I get for not even running cf-promises on it. Minor syntax errors. Fixing now.

**Comment by nickanderson** [comment]
trigger build

**Comment by kacfengine** [comment]
This is still generating crashes, for example:

```
Error Message

FAIL (UNEXPECTED FAILURE) ./17_packages/11_old/unsafe/11_packagesmatching.cf

Stacktrace

rlist.c:117: Programming Error: Internal error: Rval contains type f instead of expected scalar
Aborted
   error: Policy failed validation with command ""/home/jenkins/workspace/testing-enterprise-pr_masterfiles/label/PACKAGES_x86_64_linux_debian_6/masterfiles/tests/acceptance/workdir/__17_packages_11_old_unsafe_11_packagesmatching_cf/bin/cf-promises" -c "./17_packages/11_old/unsafe/11_packagesmatching.cf""
   error: CFEngine was not able to get confirmation of promises from cf-promises, so going to failsafe
   error: Can"t stat file "/home/jenkins/workspace/testing-enterprise-pr_masterfiles/label/PACKAGES_x86_64_linux_debian_6/masterfiles/tests/acceptance/workdir/__17_packages_11_old_unsafe_11_packagesmatching_cf/inputs/failsafe.cf" for parsing. (stat: No such file or directory)
```

See http://jenkins.usdc.cfengine.com/view/Pull%20requests/job/testing-enterprise-pr_masterfiles/40/

**Comment by tzz** [comment]
If this is caused by the variable contexts code, that works by attaching an `ifvarclass` attribute to the promise.  I don't know what specifically in the policy is causing the internal error.  @nickanderson can you find the exact line that causes the error?

I can't run this myself because unsafe tests require some special setup, sorry.

**Comment by kacfengine** [comment]
> I can't run this myself because unsafe tests require some special setup, sorry.

Many other tests also fail, such as ``promises/promises.cf``. The unsafe tests were perhaps not the best example, I didn't think about that.

**Comment by nickanderson** [comment]
It seems to be the promise in services.cf thats triggering it.

```
   reports:
-    "DEBUG|DEBUG_$(this.bundle)"::
+    DEBUG|DEBUG_classic_services::
       "DEBUG $(this.bundle): Using init system $(inits)"
       ifvarclass => and(canonify("$(inits)_set"));
```

File: lib/3.7/services.cf [hunk 1]
```diff
@@ -905,49 +905,49 @@ bundle agent classic_services(service,state)
                         canonify("$(inits)_set"));
 
   reports:
-    inform_mode::
-      "$(this.bundle): Using init system $(inits)"
+    DEBUG|DEBUG_classic_services::
+      "DEBUG $(this.bundle): Using init system $(inits)"
```
Comment (nickanderson) [inline lib/3.7/services.cf]: When the variable class guard was in use here it was causing a programming error.

**Comment by nickanderson** [comment]
trigger build

**Comment by tzz** [comment]
OK, the bug is:

```
  reports:
    "report_class_string"::
      "report" ifvarclass => and("report_class_fncall");
```

I'll look at fixing it, it's definitely not cool.

**Comment by tzz** [comment]
I believe this is fixed by https://github.com/cfengine/core/pull/2268

**Comment by jimis** [comment]
All masterfiles tests failed with
```rlist.c:117: Programming Error: Internal error: Rval contains type f instead of expected scalar```.

**Comment by tzz** [comment]
...with or without my fix?!?!?

**Comment by jimis** [comment]
without

**Comment by tzz** [comment]
Try it with my fix, the error you cite is exactly what I fixed.  It's a single library inclusion triggering the fail everywhere.

**Comment by kacfengine** [comment]
Merged to master, but not 3.7, we're too close to RC.

**Pull request merged**